### PR TITLE
feat: take interests into account when recommending activities with the assistant

### LINKED ISSUE #263: take interests into account when recommending trips

### PR DESCRIPTION
**Pull request by ioanajianu**
## Summary

This feature now allows the assistant to recommend activities taking into consideration the user's interests (the ones from their profile). This setting can be enabled or disabled by the user on the assistant screen.

## Changes

- **Screens/UI:** There is a new switch on the settings dialog of the AssistantScreen.

## Checklist

- [x] This PR resolves #263 
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached (if applicable).
- [x] Documentation has been updated (if applicable).

##  Testing

- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

## Screenshots (if applicable)

[interests.webm](https://github.com/user-attachments/assets/dffcb97d-ae41-4366-a9e6-88b800788c8c)

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt [hunk 1]
```diff
@@ -97,13 +120,14 @@ fun AssistantScreen(tripsViewModel: TripsViewModel, navigationActions: Navigatio
                           keyboardController?.hide()
                           if (uiState !is UiState.Loading) {
                             tripsViewModel.sendActivitiesPrompt(
-                                trip, prompt, provideFinalActivities)
+                                trip, prompt, interests, provideFinalActivities, useInterests)
```
Comment (eleonoraka) [inline app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt:123]: I have a small suggestion: 
Even when useInterests is false, we're still collecting and passing the interests list, though it won't be used. We could optimize this by moving the interests collection inside the sendActivitiesPrompt call, something like:

tripsViewModel.sendActivitiesPrompt(
    trip, 
    prompt, 
    if (useInterests) userViewModel.user.collectAsState().value?.interests ?: emptyList() else emptyList(),
    provideFinalActivities, 
    useInterests
)

**Review by eleonoraka — COMMENTED** [review]
While test coverage is good, consider adding a test case for when the user has no interests (empty list), I think it could contribute to make the tests more complete

File: app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt (hunk 1 above)
Comment (ioanajianu) [reply app/src/main/java/com/android/voyageur/ui/trip/assistant/Assistant.kt]: Thanks for the suggestion! Indeed, passing the interests as argument when `useInterests` was false was redundant. Instead I removed `useInterests` from the parameters, but if it's false I pass an empty list as a parameter instead of the user's interests.

File: app/src/main/java/com/android/voyageur/model/assistant/AssistantUtils.kt [hunk 2]
```diff
@@ -125,20 +131,27 @@ fun generatePrompt(trip: Trip, userPrompt: String, provideFinalActivities: Boole
           and the end date year ${endDate.first} month ${endDate.second + 1} day ${endDate.third}
             """
           .trimIndent()
+  val interestsPrompt =
+      if (interests.isNotEmpty()) {
+        "The activities should focus on the following interests (if applicable): ${interests.joinToString(", ")}."
```
Comment (eleonoraka) [inline app/src/main/java/com/android/voyageur/model/assistant/AssistantUtils.kt:136]: I have a doubt: should we put also these strings in the resources?

**Review by eleonoraka — APPROVED** [review]
Thank you for this PR! It really enriches the app
